src/models/lora.py

```diff
@@ -0,0 +1,35 @@
+from peft import LoraConfig, get_peft_model # pyright: ignore[reportPrivateImportUsage]
+from typing import Optional, Any
+from torch import nn
+
+from .transformer import TransformerModel
+from core import ContextModel
+
+class Lora(TransformerModel):
```

Comment:
should we have a `FineTuner` class that is a child of `TransformerModel`, then rename this file to finetune.py. Then make `Lora` a child of `FineTuner`. That way, we can add other finetuning methods in the future (i.e. linear probing, model-stitching, etc.)